src/OpenTK.Windowing.Desktop/Monitors.cs

```diff
@@ -22,29 +22,35 @@ namespace OpenTK.Windowing.Desktop
     /// </summary>
     public static class Monitors
     {
-        private static List<MonitorInfo> _monitorInfos = new List<MonitorInfo>();
+        static unsafe Monitors()
+        {
+            GLFWProvider.EnsureInitialized();
 
-        // I am not sure how well of an idea this is.
-        private static Dictionary<IntPtr, int> _monitorIndexLookup = new Dictionary<IntPtr, int>();
+            if (!GLFWProvider.IsOnMainThread)
+            {
+                throw new InvalidOperationException("Only GLFW main thread can build the monitor cache.");
```

Comment:
There is no cache anymore, so this exception should be replaced with "Only GLFW main thread can access this class" or something more appropriate.